drivers/wifi/eswifi/eswifi_offload.c

```diff
@@ -543,6 +580,27 @@ static int eswifi_off_get(sa_family_t family,
 		return -ENOMEM;
 	}
 
+	/* Set Transport Protocol */
+	if (ip_proto == IPPROTO_TCP) {
```

Comment:
Perhaps the compiler will optimize this properly but for the reader of the code it might be better to write this like this:
```
if (ip_proto == IPPROTO_UDP) {
	socket->type = ESWIFI_TRANSPORT_UDP;
} else if (ip_proto != IPPROTO_TCP) {
	LOG_ERR("Only TCP & UDP is supported");
	eswifi_unlock(eswifi);
	return -EPFNOSUPPORT;
}

snprintf(eswifi->buf, sizeof(eswifi->buf), "P1=%d\r", socket->type);
```
I am not sure but it looked like socket->type is initialized to ESWIFI_TRANSPORT_UDP somewhere else. If that is not the case then please adjust the code accordingly.

Reply:
@jukkar eswifi is a static instance with array of 4 sockets maximum. So the socket->type member defaulted to 0 (ESWIFI_TRANSPORT_TCP). IMO, makes sense to initialize it explicitly.

Reply:
sure, that is fine. The code could then look like this
```
if (ip_proto == IPPROTO_UDP) {
	socket->type = ESWIFI_TRANSPORT_UDP;
} else if (ip_proto == IPPROTO_TCP) {
	socket->type = ESWIFI_TRANSPORT_TCP;
} else {
	LOG_ERR("Only TCP & UDP is supported");
	eswifi_unlock(eswifi);
	return -EPFNOSUPPORT;
}

snprintf(eswifi->buf, sizeof(eswifi->buf), "P1=%d\r", socket->type);
```

Reply:
@jukkar Thanks, your are right. But I am currently checking the possibility of SSL socket option. If SSL is also adds in future, I feel switch case makes sense instead of too many if, else cases here.

Reply:
@jukkar Fixed with if, else case as of now. Will check the SSL support later. Thanks